python/paddle/optimizer/optimizer.py

```diff
@@ -120,14 +148,17 @@ def __init__(self,
                     "parameters argument given to the Optimizer should not be None in dygraph mode."
                 )
             if weight_decay is not None:
-                for param in self._parameter_list:
-                    if hasattr(param,
-                               'regularizer') and param.regularizer is not None:
-                        logging.info(
-                            "If regularizer of a Parameter has been set by 'paddle.ParamAttr' or 'static.WeightNormParamAttr' already. "
-                            "The weight_decay[%s] in Optimizer will not take effect, and it will only be applied to other Parameters!"
-                            % weight_decay.__str__())
-                        break
+                if not isinstance(self._parameter_list[0], dict):
+                    for param in self._parameter_list:
+                        if hasattr(
+                                param,
+                                'regularizer') and param.regularizer is not None:
+                            logging.info(
+                                "If regularizer of a Parameter has been set by 'paddle.ParamAttr' or 'static.WeightNormParamAttr' already. "
+                                "The weight_decay[%s] in Optimizer will not take effect, and it will only be applied to other Parameters!"
+                                % weight_decay.__str__())
+                            break
```

Comment:
For param_group, the priority of weight_decay is param.weight_decay > param_group.weight_decay > optimizer.weight_decay?

Reply:
Sure, param.weight_decay has the highest priority